finagle-redis/src/main/scala/com/twitter/finagle/redis/exp/ConnectionInitCommand.scala

```diff
@@ -0,0 +1,47 @@
+package com.twitter.finagle.redis.exp
+
+import com.twitter.finagle._
+import com.twitter.finagle.redis.ClientError
+import com.twitter.finagle.redis.param.{Database, Password}
+import com.twitter.finagle.redis.protocol._
+import com.twitter.util.Future
+
+/**
+ * A stack module that executes commands when create a new connection.
+ */
+object ConnectionInitCommand {
+
+  val Role = Stack.Role("RedisInitialCommand")
+
+  def module: Stackable[ServiceFactory[Command, Reply]] =
+    new Stack.Module2[Database, Password, ServiceFactory[Command, Reply]] {
+      val role = Role
+      val description = "Manage redis connections"
+
+      def make(db: Database, password: Password, next: ServiceFactory[Command, Reply]): ServiceFactory[Command, Reply] = {
+        for {
+          service <- next
+          authed <- auth(service, password)
+          selected <- select(authed, db)
+        } yield selected
+      }
+    }
+
+  private def select(service: Service[Command, Reply], db: Database): Future[Service[Command, Reply]] = {
+    db.index.fold(Future.value(service)) { index =>
+      service(Select(index)).flatMap {
+        case ErrorReply(msg) => Future.exception(ClientError(msg))
+        case _ => Future.value(service)
+      }
+    }
+  }
+
+  private def auth(service: Service[Command, Reply], password: Password): Future[Service[Command, Reply]] = {
+    password.code.fold(Future.value(service)) { code =>
+      service(Auth(code)).flatMap {
+        case ErrorReply(msg) => Future.exception(ClientError(msg))
+        case _ => Future.value(service)
```

Comment:
Should we look for the OK status reply?

Reply:
Auth and Select returns `StatusReply`. so handle it in https://github.com/twitter/finagle/pull/841/commits/88624eff87dbcef835267ff5f75e01095b553ccf

ref https://github.com/twitter/finagle/blob/c81a87b0cdf9105f7b2985686331d43d4c56067e/finagle-redis/src/main/scala/com/twitter/finagle/redis/ConnectionCommands.scala#L28-L48